### ISSUE TITLE
Cannot log in to OSIO; getting stuck on fabric8 login

### ISSUE THREAD

**Issue by rkratky** [openshiftio/openshift.io#2332]
Upon opening tab with OSIO, I got automatically logged out (happened same way in both Chrome and Firefox). When attempting to log back in (and after entering my log in credentials), I get redirected to:

![account-exists](https://user-images.githubusercontent.com/6623103/36535303-789d0b6e-17c9-11e8-827f-786e366f801c.png)

Clicking on **Review profile** shows:

![update-account-info](https://user-images.githubusercontent.com/6623103/36535360-a3b41f5e-17c9-11e8-86a8-f4c6f262ed09.png)

Clicking on **Submit** returns me to previous screen. On the previous screen, clicking on **Add to existing account** opens:

![fabric8-login](https://user-images.githubusercontent.com/6623103/36535423-dc4cd7d4-17c9-11e8-8e05-adda6a183224.png)

When trying to log in with my OSIO creds, I get:

![fabric8-login-invalid](https://user-images.githubusercontent.com/6623103/36535444-ef1ebe40-17c9-11e8-851d-6b24c234824c.png)

Clicking on the **Red Hat Developers** button shows the first screen again.

**Comment by sbose78** [comment]
I've investigated that this is happening for some accounts after the RHD keycloak's upgrade. 
I personally couldn't get in using my real @redhat.com account, but worked for one of my test osio accounts.

I've investigated the issue and found out that the impacted users' RHD UUID has changed. We need to check with the RHD keycloak team about this.

**Comment by sbose78** [comment]
I can confirm that logging in using the username works 
Example, if my email address is `abcd@redhat.com` , i can login using `abcd`

**Comment by sbose78** [comment]
@rkratky can you check if you are able to login using your username and not email address?

**Comment by rkratky** [comment]
@sbose78, that doesn't work for me. When I click on **LOG IN** on OSIO homepage, I get directly to the **ACCOUNT ALREADY EXISTS** page (see 1st screenshot above).

**Comment by rkratky** [comment]
Explicitly logging out of RHD and then relogging with username only works for me, too.

**Comment by joshuawilson** [comment]
Is this still a SEV1?

**Comment by sbose78** [comment]
Bumped it down.

**Comment by sbose78** [comment]
Closing because having to log in using the 'username' is a feature.

**Comment by rkratky** [comment]
> Closing

@sbose78 The behaviour is super-confusing. User getting redirected to the fabric8 pages, "account already exists", etc. -- is that all expected behaviour? I think users shouldn't see those pages at all, IMO.

> having to log in using the 'username' is a feature.

That's a weird feature because other Red Hat websites have no problems with using e-mail addresses as usernames. I mean, we can document this, but I believe it would be a mistake to disallow e-mails for logging in.

**Comment by sbose78** [comment]
@rkratky I understand the inconvenience. However, I doubt there's something that can be done to avoid this user experience - since this is a RHD side change that triggered this.  :)

**Comment by alexeykazakov** [comment]
I put Sev1 back until we fully understand what happening and what we can do with this.
Current behavior doesn't seem to be acceptable to me atm.

**Comment by sbose78** [comment]
Agreed Alexey. Seems pretty weird to me, too.

**Comment by sbose78** [comment]
Based on the update from the RHD team, it is unclear how many people are affected. I already know that a lot of redhatters have not been affected, and some have. However, the assumption is that the number would be small.

"this is mainly problem for RedHat employees as redhat automatically creates multiple accounts with same email in customer portal for employee, and even account where email is used as username is created for employee."

I would like to test out the following scenarios 

- Check if an existing non-redhat user is able to login smoothly.
- check if a new non-redhat user is able to login smoothly.
- Check if a new red hat account is able to login smoothy.

We can re-adjust severity based on the above, I think.

**Comment by cmoulliard** [comment]
If I use my "rh-gs-cmoullia" username, then I can log to "openshift.io" but
then it tells me that I'm still waiting approval

https://www.dropbox.com/s/b9kzq5282puldzh/Screenshot%202018-02-23%2019.04.50.png?dl=0

Can somebody approve my request please ?

**Comment by maxandersen** [comment]
> If I use my "rh-gs-cmoullia" username, then I can log to "openshift.io" but
> then it tells me that I'm still waiting approval

afaics your user account is cmoulliard+osio@redhat.com and that user was provisioned many months back. Was that account ever working ?

rh-gs-cmoullia looks like a complete new account ?

**Comment by cmoulliard** [comment]
I can access to openshift.io using as account `cmoulliard+osio@redhat.com`. Thanks

**Comment by alexeykazakov** [comment]
More details about this issue - https://gitlab.cee.redhat.com/dtsd/housekeeping/issues/1402

**Comment by sbose78** [comment]
This impacts OSO too .

![screenshot from 2018-02-28 19-54-33](https://user-images.githubusercontent.com/545280/36792748-6e0c68b4-1cc1-11e8-99da-a2b83785e3c7.png)

**Comment by sbose78** [comment]
@pbergene @alexeykazakov now that we are not blocked with onboarding of users, should we re-consider the severity here?

We are in touch with Vlastimil regarding his code changes - which would take at least a week.

**Comment by pbergene** [comment]
Ack, I'll leave that to Alexey - I think it would be useful to keep it as a point on the triage call regardless (if not only to satisfy my curiosity ;))

**Comment by alexeykazakov** [comment]
@rkratky RHD has been updated with a fix for this issue. Can you please verify if it now works for you when you login using your email?

**Comment by alexeykazakov** [comment]
Login should now work for majority of affected users. We still can have issues with password reset for affected users and RHD team is working on the fix for the reset password thing. I de-escalated it to Sev-2. We are not blocked anymore.

**Comment by rkratky** [comment]
@alexeykazakov It works right away in Chrome. In Firefox, I first got the following:

![weresorry](https://user-images.githubusercontent.com/6623103/36915261-5af92b32-1e50-11e8-9cc5-cceda1425baf.png)

But when I go directly to RHD and log in through there, then head back to OSIO and click on LOG IN, it goes through.

**Comment by sbose78** [comment]
Robert, that might be because of an existing session which conflicted.

For a clean test, I would suggest you to close all incognito sessions, and
then open only one incognito window - please try out the OpenShift.io login
there.

**Comment by rkratky** [comment]
@sbose78 It now works even without an incognito window. Thanks.

**Comment by kishansagathiya** [comment]
I am facing this issue on prod. Blocked on this.

**Comment by ldimaggi** [comment]
Just to confirm - you are seeing this issue TODAY? on https://openshift.io?

Was this working for you recently?

**Comment by sbose78** [comment]
Nope, this has been happening for a long long time ( since the RHD keycloak upgrade in March ). Fix was to: 
- change the password
- login user the username

**Comment by sbose78** [comment]
@mvulavak , No, once we have this merged https://github.com/fabric8-services/fabric8-auth/pull/689 - other bits will fall in place easily.

**Comment by sbose78** [comment]
I mean - yes, still an issue :)

**Comment by sbose78** [comment]
@xcoulon / @sbryzak I think the token management ( native logout support ) were the only things remaining before keycloak could be pulled out, right? https://github.com/fabric8-services/fabric8-auth/issues/350

**Comment by alexeykazakov** [comment]
@sbose78 here is the list of what left - https://github.com/openshiftio/openshift.io/issues/2772#issuecomment-453577182
